Xamarin.PropertyEditing/Resource.cs

```diff
@@ -0,0 +1,97 @@
+using System;
+
+namespace Xamarin.PropertyEditing
+{
+	public class Resource<T>
```

Comment:
One thing you can find in my branch is covariance on `T`, which is useful to handle setting a `Brush` property with a `Resource<SolidBrush>` resource. In particular, a `GetResourcesAsync (obj, brushPropertyInfo, token)`can give you all resources with a representation type that derives from `Brush`, and the client code can then manipulate those resources as `Resource<Brush>` much more easily.

Reply:
It should be unnecessary.

> which is useful to handle setting a Brush property with a Resource<SolidBrush> resource.

The type of `Resource` actually doesn't matter for setting brush properties, it only cares about the base type, the provider will then recycle it back up into a `Resource<Brush>` when it's retrieved.

>  In particular, a GetResourcesAsync (obj, brushPropertyInfo, token)can give you all resources with a representation type that derives from Brush, and the client code can then manipulate those resources as Resource<Brush> much more easily.

```csharp
Resource<SolidBrush> resource;
// Why is
IResource<Brush> basic = resource;
DoStuff (basic);
// Better than
DoStuff (resource.Value);
```

We'd be adding an interface to support something we generally don't need in the first place, I'm unconvinced.

Reply:
It's better wherever the client code is manipulating any brush, and not a concrete brush, solid or otherwise. I added it in the first place because while experimenting with using the API, it did make things easier imo. But that's fine, we can very easily add it later if we find it an improvement. That you haven't hit any issue with that PR's code shows that it's not really needed now. It's a suggestion.